selfdrive/car/hyundai/carcontroller.py

```diff
@@ -83,15 +89,49 @@ def update(self, CC, CS):
     # tester present - w/ no response (keeps relevant ECU disabled)
     if self.frame % 100 == 0 and not (self.CP.flags & HyundaiFlags.CANFD_CAMERA_SCC.value) and self.CP.openpilotLongitudinalControl:
       # for longitudinal control, either radar or ADAS driving ECU
-      addr, bus = 0x7d0, 0
-      if self.CP.flags & HyundaiFlags.CANFD_HDA2.value:
-        addr, bus = 0x730, 5
-      can_sends.append([addr, 0, b"\x02\x3E\x80\x00\x00\x00\x00\x00", bus])
+      #addr, bus = 0x7d0, 0
+      #if self.CP.flags & HyundaiFlags.CANFD_HDA2.value:
+      #  addr, bus = 0x730, 5
+      #can_sends.append([addr, 0, b"\x02\x3E\x80\x00\x00\x00\x00\x00", bus])
 
       # for blinkers
       if self.CP.flags & HyundaiFlags.ENABLE_BLINKERS:
         can_sends.append([0x7b1, 0, b"\x02\x3E\x80\x00\x00\x00\x00\x00", 5])
 
+    # # radar disable for legacy cars
+    if self.CP.openpilotLongitudinalControl:
+      self.radarDisableOverlapTimer += 1
+      self.radarDisableResetTimer = 0
+      if self.radarDisableOverlapTimer >= 30:
+        self.radarDisableActivated = True
+        if 200 > self.radarDisableOverlapTimer > 36:
+          if self.frame % 41 == 0 or self.radarDisableOverlapTimer == 37:
+            can_sends.append([0x7D0, 0, b"\x02\x10\x03\x00\x00\x00\x00\x00", 0])
+          elif self.frame % 43 == 0 or self.radarDisableOverlapTimer == 37:
+            can_sends.append([0x7D0, 0, b"\x03\x28\x03\x01\x00\x00\x00\x00", 0])
+          elif self.frame % 19 == 0 or self.radarDisableOverlapTimer == 37:
+            can_sends.append([0x7D0, 0, b"\x02\x10\x85\x00\x00\x00\x00\x00", 0])  # this disables RADAR for
```

Comment:
BTW this puts the radar into a diagnostic programming session, not sure if we want to merge this. Does it work with the 0x29 service?